src/loader.js

```diff
@@ -67,21 +67,11 @@ async function load(source) {
   // https://nextjs.org/docs/advanced-features/src-directory
   const filepath = this.resourcePath.split('pages')[1];
 
-  // Only run validation when during client compilation
-  if (!nextRuntime) {
+  // Only run validation when during server compilation
+  if (nextRuntime === 'nodejs') {
     // This is just to get subcompilation working with Next.js's fast refresh
-    let previousRefreshReg = global.$RefreshReg$;
-    let previousRefreshSig = global.$RefreshSig$;
-    let previousDocument = global.document;
-    let previousElement = global.Element;
-    // https://github.com/pmmmwh/react-refresh-webpack-plugin/issues/176#issuecomment-683150213
-    global.$RefreshReg$ = previousRefreshReg || noop;
-    global.$RefreshSig$ = previousRefreshSig || (() => noop);
-    global.document = previousDocument || {
-      querySelector: noop,
-      querySelectorAll: noop,
-    };
-    global.Element = previousElement || class Element {};
+    let previousRequire = global.require;
+    global.require = previousRequire || require || __webpack_require__;
```

Comment:
```suggestion
    global.require = previousRequire || require || __non_webpack_require__;
```
Potentially wanna use the non-webpack variant here as it's an alias for `require` x-ref: https://webpack.js.org/api/module-variables/#__non_webpack_require__-webpack-specific

Reply:
@ijjk thanks for the tip! Will this cause errors in sub-compilation if, for example, the file that is `require`d is a TypeScript file?

Reply:
Rad, thanks so much!